lib/exercism/code_analyzer.rb

```diff
@@ -0,0 +1,42 @@
+class CodeAnalyzer
+  attr_reader :language, :code, :commit_id
+
+  def initialize(options = {})
+    @language = options[:language]
+    @code = options[:code]
+    @commit_id = options[:commit_id]
+  end
+
+  def run
+    "---common---"
+  end
+
+  def self.build(options = {})
+    options[:language] ||= ""
+    if options[:language].strip.empty?
+      CodeAnalyzer.new(options)
+    else
+      const_get(options[:language].strip.capitalize).new(options) rescue CodeAnalyzer.new(options)
+    end
+  end
+end
+
+class Ruby < CodeAnalyzer
+  def run
+    unless ENV.fetch('RACK_ENV') == "test"
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.